### PR TITLE
Allow passwords and tokens to be sensitive

### PR DESCRIPTION
**Pull request by chrisst**
Notification channels are configured by sending tokens and passwords via
labels. This change allows the sensitive labels to be configured in a
separate block so that they can be marked as Sensitive and hidden from plan
output. Encoders and Decoders are used to manipulate the labels block before
it is passed to the expanders/flatteners.

The token can be specified in either place right now, but the intent is to
remove the ability to manage these from being managed via the "labels" block
in the next breaking change.

fixes https://github.com/terraform-providers/terraform-provider-google/issues/5369

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
monitoring: Added `sensitive_labels` to `google_monitoring_notification_channel` so that labels like `password` and `auth_token` can be managed separately from the other labels and marked as sensitive.
```

### CODE REVIEW

**Comment by emilymye** [comment]
there might be an issue where we dont generate sensitiveLabels in  TF conversion ([line](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-3177-old..auto-pr-3177#diff-c588a6821a0c28865c0c26fdcbfea950R91))

**Comment by chrisst** [comment]
@emilymye I don't _think_ there's an issue: The encoder should always put the value into the appropriate `labels` spot in the API request regardless of how it was specified in the config. But also I don't think the Terraform Validator [works with this resource](https://github.com/GoogleCloudPlatform/terraform-validator#resources) right now anyway so it shouldn't matter.

File: templates/terraform/encoders/monitoring_notification_channel.go.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-%>
+labelmap, ok := obj["labels"]
+if !ok {
+	labelmap = make(map[string]string)
+}
+
+var labels map[string]string
+labels = labelmap.(map[string]string)
+
+for _, sl := range sensitiveLabels {
```
Comment (emilymye) [inline templates/terraform/encoders/monitoring_notification_channel.go.erb]: can we change this to expand on v, ok := d.GetOk("sensitive_labels")?

File: third_party/terraform/tests/resource_monitoring_notification_channel_test.go [hunk 2]
```diff
@@ -7,27 +7,6 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 )
 
-func TestMonitoringNotificationChannel_labelsObfuscated(t *testing.T) {
```
Comment (emilymye) [inline third_party/terraform/tests/resource_monitoring_notification_channel_test.go]: should we also add an example/test for the sensitive label

**Comment by emilymye** [comment]
@chrisst i meant the constants arent rendered in conversion, so that variable doesnt exist and it wont build

**Comment by chrisst** [comment]
Ah nuts, I'll fix the constants thanks.

File: templates/terraform/encoders/monitoring_notification_channel.go.erb (hunk 1 above)
Comment (chrisst) [reply templates/terraform/encoders/monitoring_notification_channel.go.erb]: If you are talking about line 23 it's just iterating through the constant slice of possible labels to check for.
If you are talking about line 24, are you wanting to update the variable names or GetOk vs GetOkExists?

**Pull request merged**